transport/src/commonMain/kotlin/com/genesys/cloud/messenger/transport/core/CustomAttributesStoreImpl.kt

```diff
@@ -1,25 +1,59 @@
 package com.genesys.cloud.messenger.transport.core
 
+import com.genesys.cloud.messenger.transport.core.events.Event
+import com.genesys.cloud.messenger.transport.core.events.EventHandler
 import com.genesys.cloud.messenger.transport.util.logs.Log
+import io.ktor.utils.io.charsets.Charsets
+import io.ktor.utils.io.core.toByteArray
 
-internal class CustomAttributesStoreImpl(private val log: Log) : CustomAttributesStore {
+internal class CustomAttributesStoreImpl(
+    private val log: Log,
+    private val eventHandler: EventHandler,
+) : CustomAttributesStore {
     private var customAttributes: MutableMap<String, String> = mutableMapOf()
+    var maxCustomDataBytes: Int = 0
+
     internal var state: State = State.PENDING
         private set
 
     override fun get(): Map<String, String> = customAttributes
 
-    override fun add(customAttributes: Map<String, String>) {
-        if (customAttributes.isEmpty() || this.customAttributes == customAttributes) return
-        state = State.PENDING
-        log.i { "add: $customAttributes | state = $state" }
-        for ((key, value) in customAttributes) {
-            this.customAttributes[key] = value
+    override fun add(customAttributes: Map<String, String>): Boolean {
+        return if (customAttributes.isEmpty() || this.customAttributes == customAttributes) {
+            log.i { "custom attributes are empty are same." }
+            false
+        } else if (isSizeExceeded(customAttributes)) {
+            eventHandler.onEvent(
+                Event.Error(
+                    ErrorCode.CustomAttributeSizeTooLarge,
+                    ErrorMessage.customAttributesSizeError(maxCustomDataBytes),
+                    CorrectiveAction.CustomAttributeSizeTooLarge
+                )
+            )
+            log.e { "error: custom attributes size exceeded" }
+            false
+        } else {
+            this.customAttributes.putAll(customAttributes)
+            state = State.PENDING
+            log.i { "add: $customAttributes | state = $state" }
+            true
         }
     }
 
-    internal fun getCustomAttributesToSend(): Map<String, String> =
-        if (state == State.PENDING) get() else emptyMap()
+    private fun isSizeExceeded(attributes: Map<String, String>): Boolean {
+        val totalSize = attributes.entries.sumOf { (key, value) ->
+            key.toByteArray(Charsets.UTF_8).size + value.toByteArray(Charsets.UTF_8).size
+        }
+        return totalSize > maxCustomDataBytes
```

Comment:
Worth checking if totalSize should be `>` or `>=`.

Reply:
The documentation says custom attributes have to be less than 2kb(2048 bytes).